### PR TITLE
Apply spec: Added support for NFS and MDS service

### PR DESCRIPTION
**Pull request by sunilkumarn417**
- NFS and MDS service deployment using spec file.

**Test results:**
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1620808639565

### CODE REVIEW

File: ceph/ceph_admin/helper.py [hunk 1]
```diff
@@ -200,6 +200,73 @@ def generate_osd_spec(self, spec):
...
+        Returns:
+            service_spec
+        """
+        template = self._get_template("mds")
+        node_names = spec["placement"].pop("nodes", None)
+        if node_names:
+            spec["placement"]["hosts"] = []
+            nodes = get_nodes_by_ids(self.cluster, node_names)
+            for node in nodes:
+                spec["placement"]["hosts"].append(node.shortname)
```
Comment (psathyan) [inline ceph/ceph_admin/helper.py:231]: Would it make sense to have a separate method to return the host list which then can we set to `spec["placement"]["hosts"]`? 

I noticed that we are using these lines across multiple methods.

**Review by psathyan — COMMENTED** [review]
Looks good, a minor improvement.

**Comment by sunilkumarn417** [comment]
**latest results:**
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1621224191694/result.html

File: ceph/ceph_admin/helper.py (hunk 1 above)
Comment (sunilkumarn417) [reply ceph/ceph_admin/helper.py]: Done.

**Review by psathyan — APPROVED** [review]

**Comment by rakeshgm** [comment]
I believe this should be nfs ganesha or nfs ganesha rgw and not just nfs we will also be having nfs for Cephfs too. @sunilkumarn417 can you refactor the verbiage

**Comment by sunilkumarn417** [comment]
> I believe this should be nfs ganesha or nfs ganesha rgw and not just nfs we will also be having nfs for Cephfs too. @sunilkumarn417 can you refactor the verbiage

Since the way NFS is deployed is common, did not wanted to name it with protocol. However the test case is defined with protocol.

**Pull request merged**